### PR TITLE
Build cmd and additional updates for VS upgrade

### PR DESCRIPTION
**Pull request by Sreejithpin**
### Issues

Updates of Visual studio upgrade project

*This pull request fixes issue #xxx.*

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: Build.props [hunk 1]
```diff
@@ -67,6 +67,11 @@
     <!--Overide the default full debug symbols regardless of configuration-->
     <DebugType>pdbonly</DebugType>
   </PropertyGroup>
+
+  <PropertyGroup Condition="'$(MSBuildProjectExtension)' == '.vbproj'">
```
Comment (xuzhg) [inline Build.props:71]: why do we need to process "vb" again?

File: README.md [hunk 2]
```diff
@@ -23,11 +23,12 @@ For more information about OData, please refer to the following resources:
...
+This master branch is the development branch for ODataV4 7.x and is now most actively iterated. It builds upon the OData 7.6 release which is now on [master_vs2015_maintenance](https://github.com/OData/odata.net/tree/master_vs2015_maintenance) and produces both [PCL (Portable Class Libraries) Profile111](https://msdn.microsoft.com/library/gg597391.aspx) and [.NET Standard 1.1](https://docs.microsoft.com/en-us/dotnet/articles/standard/library) libraries. The branch builds mostly with Visual Studio 2015, and it is undergoing migration towards Visual Studio 2017; currently, the .NET Standard and .NET Core projects have been migrated. Due to the number of test projects, complete migration to the latest version of Visual Studio will be broken down into multiple steps. The code is shared between the .NET Framework and .NET Standard platforms, and you may use either Visual Studio 2015/2017 to contribute.
```
Comment (xuzhg) [inline README.md:30]: VS2019 is recommended to build master branch, right? why do we say "2015/2017" ?

File: README.md [hunk 3]
```diff
@@ -81,15 +82,8 @@ Simply open the shortcut `OData.sln` at the root level folder to launch a soluti
...
-- OData.Tests.NetStandard.VS2017.sln - Product source built with .Net Standard 1.1 and contains corresponding unit tests written in .NET Core. Note that once OData has migrated to VS2017, this solution will replace OData.NetStandard.sln (and take its name) to reduce the number of solutions.
-- OData.Tests.Performance.sln - Product source and performance tests built with .Net Framework version 4.6.
-- OData.Tests.WindowsApps.sln - Product source and test harnesses written in Windows Portable and Windows Phone 8.1.
+- OData.sln - Main OData Solution with Product and unit test.
+- OData.E2E.sln - Product source built with .Net Framework 4.5. Contains exhaustive list of tests (unit, E2E, and regression) and not intended to be opened frequently. The `Build.cmd` script will run all tests from here and this solution is used to _fully_ test your code.
 
 Each solution contains some test projects. Please open it, build it and run all the tests in the test explorer. For running tests within OData.Tests.E2E.sln and OData.Tests.E2E.NetCore.VS2017.sln, you need to open Visual Studio IDE as **_Administrator_** so that the test services can be started properly.
```
Comment (xuzhg) [inline README.md:88]: Change the NetCore.VS2017.sln ?

Comment (xuzhg) [reply README.md:88]: Meanwhile, list the potential build error and our workaround. for example the long path limitation?
the build warnings?

File: azure-pipelines-nightly.yml [hunk 4]
```diff
@@ -205,49 +205,50 @@ steps:
      popd
     preferredAzurePowerShellVersion: 5.1.1
 
+
 - task: DotNetCoreCLI@2
   displayName: 'Spatial Test'
   inputs:
     command: 'test'
-    arguments: '--configuration $(BuildConfiguration) --no-build'
```
Comment (xuzhg) [inline azure-pipelines-nightly.yml:212]: why to run a certain collection, not the whole test project?

File: build.ps1 [hunk 5]
```diff
@@ -55,12 +55,12 @@ $env:ENLISTMENT_ROOT = Split-Path -Parent $MyInvocation.MyCommand.Definition
...
-# Default to use Visual Studio 2015
-$VS14MSBUILD=$PROGRAMFILESX86 + "\MSBuild\14.0\Bin\MSBuild.exe"
-$VSTEST = $PROGRAMFILESX86 + "\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe"
+# Default to use Visual Studio 2019 C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin
+$VS14MSBUILD=$PROGRAMFILESX86 + "\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe"
+$VSTEST = $PROGRAMFILESX86 + "\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe"
 $FXCOPDIR = $PROGRAMFILESX86 + "\Microsoft Visual Studio 14.0\Team Tools\Static Analysis Tools\FxCop"
-$SN = $PROGRAMFILESX86 + "\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\sn.exe"
-$SNx64 = $PROGRAMFILESX86 + "\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\x64\sn.exe"
+$SN = $PROGRAMFILESX86 + "\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6.1 Tools\sn.exe"
```
Comment (xuzhg) [inline build.ps1:62]: Does other developer has such tool installed?

File: build.ps1 [hunk 6]
```diff
@@ -88,14 +88,15 @@ if ([System.IO.File]::Exists($DOTNETDIR + "dotnet.exe"))
 $FXCOP = $FXCOPDIR + "\FxCopCmd.exe"
 $BUILDLOG = $LOGDIR + "\msbuild.log"
 $TESTLOG = $LOGDIR + "\mstest.log"
-$TESTDIR = $ENLISTMENT_ROOT + "\bin\AnyCPU\$Configuration\Test\Desktop"
-$NETCORETESTDIR = $ENLISTMENT_ROOT + "\bin\AnyCPU\$Configuration\Test\.NETPortable\netcoreapp1.0"
-$PRODUCTDIR = $ENLISTMENT_ROOT + "\bin\AnyCPU\$Configuration\Product\Desktop"
+$TESTDIR = $ENLISTMENT_ROOT + "\bin\AnyCPU\$Configuration\Test\Desktop\net452"
```
Comment (xuzhg) [inline build.ps1:91]: we should remove "Desktop"

File: build.ps1 [hunk 7]
```diff
@@ -88,14 +88,15 @@ if ([System.IO.File]::Exists($DOTNETDIR + "dotnet.exe"))
...
+$TESTDIRE2E = $ENLISTMENT_ROOT + "\bin\AnyCPU\$Configuration\Test\Desktop\net45"
+$NETCORETESTDIR = $ENLISTMENT_ROOT + "\bin\AnyCPU\$Configuration\Test\.NETPortable\netcoreapp3.1"
```
Comment (xuzhg) [inline build.ps1:93]: remove ".NETPortable"

File: build.ps1 [hunk 8]
```diff
@@ -88,14 +88,15 @@ if ([System.IO.File]::Exists($DOTNETDIR + "dotnet.exe"))
...
+$PRODUCTDIR = $ENLISTMENT_ROOT + "\bin\AnyCPU\$Configuration\Product\Desktop\net45"
 $NUGETEXE = $ENLISTMENT_ROOT + "\sln\.nuget\NuGet.exe"
 $NUGETPACK = $ENLISTMENT_ROOT + "\sln\packages"
 $XUNITADAPTER = "/TestAdapterPath:" + $NUGETPACK + "\xunit.runner.visualstudio.2.1.0\build\_common"
 
-$NugetRestoreSolutions = "OData.NetStandard.sln"
+$NugetRestoreSolutions = "OData.sln"
```
Comment (xuzhg) [inline build.ps1:99]: We don't need to "NugetRestore" again. 
dotnet.exe build do the nuget restore automatically\
So, consider to remove this part.

File: build.ps1 [hunk 9]
```diff
@@ -567,11 +567,12 @@ Function TestProcess
...
     if ($TestType -eq 'Nightly')
     {
-        RunTest -title 'NightlyTests' -testdir $NightlyTestSuite
+        RunTest -title 'NetCoreUnitTests' -testdir $NetCoreXUnitTestProjs -framework 'dotnet'
+		RunTest -title 'NetCoreE2ETests' -testdir $NetCoreE2ETestProjs -framework 'dotnet'
     }
     elseif ($TestType -eq 'Quick')
-    {
-        RunTest -title 'XUnitTests' -testdir $QuickTestSuite
+    {	Write-Output "==Running Quick Test 9999999 =="
```
Comment (xuzhg) [inline build.ps1:574]: a new line

File: build.ps1 [hunk 10]
```diff
@@ -582,7 +583,10 @@ Function TestProcess
 
     if ($DOTNETTEST)
     {
-        RunTest -title 'NetCoreTests' -testdir $NetCoreXUnitTestProjs -framework 'dotnet'
+		Write-Output "==Running Quick Test 00000 =="
```
Comment (xuzhg) [inline build.ps1:586]: align

**Comment by xuzhg** [comment]
@Sreejithpin 
1) make the PR title meaningful
2) if you think it's ready, label the PR "ready for review".
3) Can you have a clean VM to test the build.cmd?

File: build.ps1 (hunk 5 above)
Comment (Sreejithpin) [reply build.ps1:62]: Was there already , so i assume so

File: Build.props (hunk 1 above)
Comment (Sreejithpin) [reply Build.props]: There is 1-2 vb projects, can we remove all vb projects from e2e? Those are not in build earlier also, can we remove from solution/ any references

Comment (mikepizzo) [reply Build.props]: Are these just test projects, testing the libraries from VB?

Comment (Sreejithpin) [reply Build.props]: Yes E2e test for vb, but those were/are not in ADO, should we remove those projects

File: azure-pipelines-nightly.yml (hunk 4 above)
Comment (Sreejithpin) [reply azure-pipelines-nightly.yml:212]: collect code cov for each project

**Comment by mikepizzo** [comment]
@Sreejith  - What's the status of this PR?

File: README.md (hunk 2 above)
Comment (Sreejithpin) [reply README.md]: Readme is fixed altogether

File: README.md (hunk 3 above)
Comment (Sreejithpin) [reply README.md]: Readme is fixed altogether

File: README.md [hunk 11]
```diff
@@ -1,161 +1 @@
...

### Thank you

We’re using NDepend to analyze and increase code quality.

[![NDepend](images/ndependlogo.png)](http://www.ndepend.com)

### Code of Conduct

This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
```
Comment (odero) [inline README.md:1]: The readme has a lot of weird characters could you fix those?

Comment (Sreejithpin) [reply README.md]: Fixed

**Pull request merged**